src/components/Amalgamation/AmalgamatingBusinesses.vue

```diff
@@ -0,0 +1,255 @@
+
+  // Add an Amalgamating Business button properties
+  isAddingAmalgamatingBusiness = false
+  addAmalgatingBusinessDisabled = false
+
+  // Add an Amalgamating Foreign Business button properties
+  isAddingAmalgamatingForeignBusiness = false
+  addAmalgatingForeignBusinessDisabled = false
+
+  readonly BusinessLookupServices = BusinessLookupServices
+
+  // Cancel button in "Add an Amalgamating Business" is pressed.
+  addAmalgamatingBusinessCancel (): void {
+    this.isAddingAmalgamatingBusiness = false
+    // Enable buttons
+    this.addAmalgatingBusinessDisabled = false
+    this.addAmalgatingForeignBusinessDisabled = false
+  }
+
+  // "Add an Amalgamating Business" button is pressed.
+  onAddBusinessClick (): void {
+    this.isAddingAmalgamatingBusiness = true
+    this.isAddingAmalgamatingForeignBusiness = false
+    // Disable buttons
+    this.addAmalgatingBusinessDisabled = true
+    this.addAmalgatingForeignBusinessDisabled = true
+  }
+
+  // "Add an Amalgamating Foreign Business" button is pressed.
+  onAddForeignBusinessClick (): void {
+    this.isAddingAmalgamatingBusiness = false
+    this.isAddingAmalgamatingForeignBusiness = true
+    // Disable buttons (Please comment out the lines below or remove in part 2)
+    // this.addAmalgatingBusinessDisabled = true
+    // this.addAmalgatingForeignBusinessDisabled = true
+  }
+
+  // Add to the amalgamating businesses array.
+  // If EP (A type), cannot be part of short form horizontal amalgamation.
+  pushToAmalgamatingBusinesses (business: any): void {
+    if (this.isAmalgamationFilingHorizontal) {
+      if (business.legalType !== 'A') this.amalgamatingBuinesses.push(business)
+    } else {
+      this.amalgamatingBuinesses.push(business)
+    }
+  }
+
+  async setAmalgamatingBusiness (businessLookup: BusinessLookupIF): Promise<void> {
+    // Get the amalgamating business information
+    // Will have a different format depending on the business
+    let business = await LegalServices.fetchBusinessInfo(businessLookup.identifier)
+      .then((response) => {
+        return response?.data?.business
+      }).catch(() => {
+        return businessLookup
+      })
+
+    // Get the address of the amalgamating business
+    if (businessLookup.identifier && business.foundingDate) {
+      const addresses = await LegalServices.fetchAddresses(businessLookup.identifier)
+        .then((data) => {
+          // SP and GP have businessOffice instead of registeredOffice
+          return data?.registeredOffice || data?.businessOffice
+        }).catch(() => {
+          return undefined
+        })
+      if (addresses) {
+        business.officeAddress = addresses
+      }
+    }
+
+    // If the amalgamating businesses array is not empty, check if identifier already exists.
+    // If identifier already exists, don't add the business to the array.
+    if (this.amalgamatingBuinesses.length > 0) {
+      const filteredBusinesses = this.amalgamatingBuinesses.filter(function (id) {
```

Comment:
Use `find` instead. (Find returns on first match, while filter searches the whole list.)